### PR TITLE
[auth] Factor out KeyResolver

### PR DESCRIPTION
**Pull request by vosst**
This PR factors out an interface `auth.KeyResolver` and adjusts the existing authorizer to use implementations of that interface for initializing and refreshing cryptographic keys for JWT verification. With that , we can point a DSS instance to a well-known endpoint serving JWKS and add support for transparently cycling JWT signing keys.

### CODE REVIEW

File: pkg/dss/auth/auth_test.go [hunk 1]
```diff
@@ -38,10 +38,19 @@ func rsaTokenCtx(ctx context.Context, key *rsa.PrivateKey, exp, nbf int64) conte
...
 
 func TestNewRSAAuthClient(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	tmpfile, err := ioutil.TempFile("/tmp", "bad.pem")
 	require.NoError(t, tmpfile.Close())
 	// Test catches previous segfault.
-	_, err = NewRSAAuthClient(tmpfile.Name(), nil)
+	// NewRSAAuthorizer returns an Authorizer instance using
```
Comment (davidsansome) [inline pkg/dss/auth/auth_test.go]: I think you forgot to finish the comment!

File: pkg/dss/auth/auth_test.go [hunk 2]
```diff
@@ -73,7 +84,15 @@ func TestRSAAuthInterceptor(t *testing.T) {
 		{rsaTokenCtx(ctx, key, 100, 50), codes.Unauthenticated}, // Not valid yet
 	}
 
-	a := &authClient{key: &key.PublicKey, logger: zap.L()}
+	// NewRSAAuthorizer returns an Authorizer instance using
```
Comment (davidsansome) [inline pkg/dss/auth/auth_test.go]: Same here

File: pkg/dss/auth/auth.go [hunk 3]
```diff
@@ -45,40 +54,154 @@ func OwnerFromContext(ctx context.Context) (models.Owner, bool) {
...
 }
 
-type authClient struct {
-	logger           *zap.Logger
-	key              interface{}
-	requiredScopes   map[string][]string
-	requiredAudience string
+// KeyResolver abstracts resolving keys.
+type KeyResolver interface {
+	// ResolveKey resolves a key for verifying JWTs.
```
[the remark this answers is not in the document]
Comment (vosst) [reply pkg/dss/auth/auth.go]: Fixed!

File: pkg/dss/auth/auth_test.go (hunk 1 above)
Comment (vosst) [reply pkg/dss/auth/auth_test.go]: Removed, it was a c'n'p artifact anyway.

File: pkg/dss/auth/auth_test.go (hunk 2 above)
Comment (vosst) [reply pkg/dss/auth/auth_test.go]: Same here, c'n'p artifact.

**Pull request merged**